modules/sdk-coin-ada/src/ada.ts

```diff
@@ -207,6 +232,159 @@ export class Ada extends BaseCoin {
     };
   }
 
+  protected getPublicNodeUrl(): string {
+    return Environments[this.bitgo.getEnv()].adaNodeUrl;
+  }
+
+  protected async getDataFromNode(endpoint: string, requestBody?: Record<string, unknown>): Promise<request.Response> {
+    const restEndpoint = this.getPublicNodeUrl() + '/' + endpoint;
+    try {
+      const res = await request.post(restEndpoint).send(requestBody);
+      return res;
+    } catch (e) {
+      console.debug(e);
+    }
+    throw new Error(`Unable to call endpoint ${restEndpoint}`);
+  }
+
+  protected async getAddressInfo(
+    walletAddr: string
+  ): Promise<{ balance: number; utxoSet: Array<Record<string, any>> }> {
+    const requestBody = { _addresses: [walletAddr] };
+    const res = await this.getDataFromNode('address_info', requestBody);
+    if (res.status != 200) {
+      throw new Error(`Failed to retrieve address info for address ${walletAddr}`);
+    }
+    const body = res.body[0];
+    return { balance: body.balance, utxoSet: body.utxo_set };
+  }
+
+  protected async getChainTipInfo(): Promise<Record<string, string>> {
+    const res = await this.getDataFromNode('tip');
+    if (res.status != 200) {
+      throw new Error('Failed to retrieve chain tip info');
+    }
+    const body = res.body[0];
+    return body;
+  }
+
+  /**
+   * Builds a funds recovery transaction without BitGo
+   *
+   * @param {RecoveryOptions} params parameters needed to construct and
+   * (maybe) sign the transaction
+   *
+   * @returns {AdaTx} the serialized transaction hex string and index
+   * of the address being swept
+   */
+  async recover(params: RecoveryOptions): Promise<AdaTx> {
+    if (!params.bitgoKey) {
+      throw new Error('missing bitgoKey');
+    }
+    if (!params.recoveryDestination || !this.isValidAddress(params.recoveryDestination)) {
+      throw new Error('invalid recoveryDestination');
+    }
+    let addrFormat;
+    if (Ada.TestnetEnv.has(this.bitgo.getEnv())) {
```

Comment:
can we just do regular compare here?

Reply:
The reason why I did it like this is because I thought there will be multiple environment that will correspond to using either testnet or mainnet. If the list grows in the future doing it with regular comparison could look ugly.